include/sprite.h

```diff
@@ -151,6 +151,26 @@ enum
     SUBSPRITES_IGNORE_PRIORITY, // on but priority is ignored
 };
 
+#define OAM_SIZE_0 (0)
+#define OAM_SIZE_1 (1 << 0)
+#define OAM_SIZE_2 (1 << 1)
+#define OAM_SIZE_3 (1 << 2)
+
+#define SPRITE_SIZE_8x8 .shape = ST_OAM_SQUARE, .size = OAM_SIZE_0
```

Comment:
I like this.

Reply:
Hmm, actually I don't really like the current use of it, since it's hiding two fields when making the struct definition, and look rather out of place to me.  I'd rather see something like this, I think:
```
.shape = OAM_SHAPE(SHAPE_32x64),
.size = OAM_SIZE(SIZE_32x64),
// or maybe
.shape = OAM_SHAPE(32, 64),
.size = OAM_SIZE(32, 64),
```

Plus these should be used in `struct OamData` definitions, but this PR is not actually using them.

Reply:
Yeah, I was iffy about the details.
Oh, that's a good idea. Combining them and then splitting them like we do for map numbers / map groups.

And I didn't do it for Oam Data stuff specifically because I wasn't happy about it hiding two fields. I can update all the Oam stuff to use the new system, but that's probably best saved as a separate PR.

Reply:
how do you feel about these? they would require people to understand the allowable GBA sprite dimensions because you can't just put any number for width and height and expect good behavior, 
```
#define OAM_SHAPE(width, height) \
    (width == height ? ST_OAM_SQUARE : \
    (width < height ? ST_OAM_V_RECTANGLE : ST_OAM_H_RECTANGLE))

#define OAM_SIZE(width, height) \
    (width * height <= 128 ? OAM_SIZE_0 : \
    (width * height == 256 ? OAM_SIZE_1 : \
    (width * height <= 1024 ? OAM_SIZE_2 : OAM_SIZE_3))

```

Reply:
Oh, I've already solved this problem and updated this branch. I must have forgotten to push.

I did it almost like that.